### PR TITLE
fix: Make empty role=lisbox elements as incomplete

### LINKED ISSUE #874: "aria-required-children" failure on role "listbox" for missing child nodes in the DOM
```aria-required-children``` fails for a ```listbox``` not having a child role ```option```. 
In a few cases authors tend to create the child ```option```elements on the fly. 

So, on roles like ```listbox``` should we tweak ```aria-required-children``` rule to:
- Needs Review: If a ```listbox``` has no child elements that are exposed to the ATs
- Fail: If a ```listbox``` has child elements exposed to ATs and is missing role ```option```

Link to live demo: https://www.bing.com/

<pre><code>axe-core version: 3.0.0-beta.3
axe-Coconut version: 3.3.0-beta.0
</code></pre>

### PR DESCRIPTION
**Pull request by WilcoFiers**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message(s) follow our guidelines: https://github.com/dequelabs/axe-core/blob/develop/doc/code-submission-guidelines.md#git-commits
- [x] Changes to rules and checks appropriately support [Shadow DOM](https://github.com/dequelabs/axe-core/blob/develop/doc/developer-guide.md)
- [x] Changes have been tested in [major browsers and Assistive Technologies](https://github.com/dequelabs/axe-core/blob/develop/doc/accessibility-supported.md#accessibility-supported)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Description of the changes

- Github issue: Closes #874

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: lib/checks/aria/required-children.js [hunk 1]
```diff
@@ -86,4 +87,10 @@ var missing = missingRequiredChildren(node, childRoles, all, role);
 if (!missing) { return true; }
 
 this.data(missing);
-return false;
+
+console.log(role, reviewEmpty.includes(role), missing);
```
Comment (jkodu) [inline lib/checks/aria/required-children.js]: left intentionally?

Comment (WilcoFiers) [reply lib/checks/aria/required-children.js]: Nope. Why did you approve?

Comment (jkodu) [reply lib/checks/aria/required-children.js]: No harm, hence...

**Comment by dylanb** [comment]
@WilcoFiers shouldn't we just pass these, rather than making them a review item? An empty list is a valid list after all.

**Comment by dylanb** [comment]
I think it is actually an oversight on the part of ARIA. Even their own examples have this issue.

File: lib/checks/aria/required-children.json [hunk 2]
```diff
@@ -1,11 +1,15 @@
 {
   "id": "aria-required-children",
   "evaluate": "required-children.js",
+  "options": {
+    "reviewEmpty": ["listbox"]
```
Comment (BrendanMcK) [inline lib/checks/aria/required-children.json]: Should allow 'list' also?

**Pull request merged**